classes/Cart.php

```diff
@@ -3786,11 +3786,23 @@ public function checkDiscountValidity($obj, $discounts, $order_total, $products,
     }
 
     /**
-     * Return useful information about the cart.
+     * Return useful information about the cart for display purpose.
      *
      * @return array Cart details
      */
     public function getSummaryDetails($id_lang = null, $refresh = false)
+    {
+        $summary = $this->getUnalteredSummaryDetails($id_lang, $refresh);
+
+        return $this->alterSummaryForDisplay($summary, $refresh);
+    }
+
+    /**
+     * Returns useful raw informations about the cart.
+     *
+     * @return array Cart details
+     */
+    public function getUnalteredSummaryDetails($id_lang = null, $refresh = false): array
```

Comment:
```suggestion
    public function getRawSummaryDetails($id_lang = null, $refresh = false): array
```

Reply:
Since we're at it, can I suggest making `id_lang` not optional?
```suggestion
    public function getRawSummaryDetails(int $id_lang, $refresh = false): array
```

Reply:
We have some use cases xhere id_lang is not given. In these cases the id_lang is retrieved from the context
Do you suggest we give directly the lang in the context as parameter to the call ?

Reply:
You can keep the fallback in getSummaryDetails to allow a clean interface (i.e. make the parameter required required) for getRawSummaryDetails :)